api/lagoon/client/deploytargets.go

```diff
@@ -59,59 +59,44 @@ func (c *Client) DeleteDeployTarget(ctx context.Context, in *schema.DeleteDeploy
 	return c.client.Run(ctx, req, &out)
 }
 
-// DeployTargetsByOrganizationId queries the Lagoon API for deploy targets by the given organization id
+// DeployTargetsByOrganizationNameOrID queries the Lagoon API for deploy targets by the given organization name
 // and unmarshals the response into organization.
-func (c *Client) DeployTargetsByOrganizationId(ctx context.Context, id uint, out *[]schema.DeployTarget) error {
-
-	req, err := c.newRequest("_lgraphql/deploytargets/deployTargetsByOrganizationId.graphql",
-		map[string]interface{}{
-			"id": id,
-		})
-	if err != nil {
-		return err
-	}
-
+// Need to rework the argument types here
+func (c *Client) DeployTargetsByOrganizationNameOrID(ctx context.Context, name *string, id *uint, out *[]schema.DeployTarget) error {
+	var req *graphql.Request
+	var err error
 	o := &schema.Organization{}
-	err = c.client.Run(ctx, req, &struct {
-		Response *schema.Organization `json:"organizationById"`
-	}{
-		Response: o,
-	})
-	if err != nil {
-		return err
-	}
-	if len(o.DeployTargets) == 0 {
-		return fmt.Errorf("no deploy targets found for organization %s", o.Name)
-	}
-	data, err := json.Marshal(o.DeployTargets)
-	if err != nil {
-		return err
-	}
-	json.Unmarshal(data, out)
-	return nil
-}
-
-// DeployTargetsByOrganizationName queries the Lagoon API for deploy targets by the given organization name
-// and unmarshals the response into organization.
-func (c *Client) DeployTargetsByOrganizationName(ctx context.Context, name string, out *[]schema.DeployTarget) error {
 
-	req, err := c.newRequest("_lgraphql/deploytargets/deployTargetsByOrganizationName.graphql",
-		map[string]interface{}{
-			"name": name,
+	if *name != "" {
```

Comment:
Check for `nil`, same for `id`
```
	if name != nil {
		req, err = c.newRequest("_lgraphql/deploytargets/deployTargetsByOrganizationName.graphql",
			map[string]interface{}{
				"name": *name,
			})
```

Reply:
Should have been updated with the previous commit

Reply:
Ah yep, my bad. Still something to fix in the update though